### PR TITLE
Cast `inputs` array to float64 in case of big endian architecture for normalization

### LINKED ISSUE #26135: //tensorflow/contrib/layers:normalization_test fails with Assertion error
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):     Linux Ubuntu 16.04 s390x
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): 1.12.0
- Python version: 2.7.12
- Bazel version (if compiling from source): 0.15.0
- GCC/Compiler version (if compiling from source): gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.11)
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the current behavior**
python tensorflow/contrib/layers/python/layers/normalization_test.py fails on s390x with following parameters
input_shape = (1, 100, 100, 1)
data_format =  'NCHW' and 
tolerance = 1e-3
mu = 1e2
sigma = 0.1

**Describe the expected behavior**
It should pass on s390x like on Intel.

**Code to reproduce the issue**
python tensorflow/contrib/layers/python/layers/normalization_test.py Or
bazel test -c dbg  //tensorflow/contrib/layers:normalization_test

root@1da9d9e529b7:/home/test/tensorflow# python tensorflow/contrib/layers/python/layers/normalization_test.py
..name:  GroupNorm/Reshape_2
......................name:  InstanceNorm/instancenorm/add_1
....F.........
FAIL: testOutputBigInput4DNCHW (__main__.InstanceNormTest)

Traceback (most recent call last):
  File "tensorflow/contrib/layers/python/layers/normalization_test.py", line 172, in testOutputBigInput4DNCHW
    self.doOutputTest((1, 100, 100, 1), 'NCHW', tol=1e-3)
  File "tensorflow/contrib/layers/python/layers/normalization_test.py", line 155, in doOutputTest
    self.assertAllClose(expected_mean, mean, rtol=tol, atol=tol)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 1591, in assertAllClose
    self._assertAllCloseRecursive(a, b, rtol=rtol, atol=atol, msg=msg)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 1561, in _assertAllCloseRecursive
    (path_str, path_str, msg)))
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/test_util.py", line 1496, in _assertArrayLikeAllClose
    a, b, rtol=rtol, atol=atol, err_msg="\n".join(msgs), equal_nan=True)
  File "/usr/local/lib/python2.7/dist-packages/numpy/testing/utils.py", line 1395, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/usr/local/lib/python2.7/dist-packages/numpy/testing/utils.py", line 778, in assert_array_compare
    raise AssertionError(msg)
AssertionError:
Not equal to tolerance rtol=0.001, atol=0.001
Mismatched value: a is different from b.
not close where = (array([0, 0, 0, 0, 0, 0]), array([ 8, 34, 44, 78, 87, 89]))
not close lhs = [ 0.  0.  0.  0.  0.  0.]
not close rhs = [-0.00105469 -0.00128662  0.00124512  0.00106445 -0.00124512 -0.00104004]
not close dif = [ 0.00105469  0.00128662  0.00124512  0.00106445  0.00124512  0.00104004]
not close tol = [ 0.00100105  0.00100129  0.00100125  0.00100106  0.00100125  0.00100104]
dtype = float64, shape = (1, 100)
(mismatch 6.0%)
 x: array([[ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,
         0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,
         0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,...
 y: array([[  2.612305e-04,   1.684570e-04,  -1.342773e-04,  -1.684570e-04,
         -7.495117e-04,  -3.417969e-05,  -5.371094e-05,   6.567383e-04,
         -1.054688e-03,   2.319336e-04,  -1.318359e-04,  -3.222656e-04,...

----------------------------------------------------------------------
Ran 38 tests in 16.724s

FAILED (failures=1)

### PR DESCRIPTION
**Pull request by kbhute-ibm**
These changes fixes https://github.com/tensorflow/tensorflow/issues/26135

### CODE REVIEW

**Comment by kbhute-ibm** [comment]
Hi @martinwicke Gentle reminder to merge this PR.

**Review by martinwicke — DISMISSED** [review]

File: tensorflow/contrib/layers/python/layers/normalization.py [hunk 1]
```diff
@@ -93,6 +94,9 @@ def instance_norm(inputs,
   inputs = ops.convert_to_tensor(inputs)
   inputs_shape = inputs.shape
   inputs_rank = inputs.shape.ndims
+  # Cast `inputs` array to float64 if machine is big endian
```
Comment (martinwicke) [inline tensorflow/contrib/layers/python/layers/normalization.py]: This code comment could be better. Reading the code, I see what this does, but I don't know why. Could you expand on that?

Comment (kbhute-ibm) [reply tensorflow/contrib/layers/python/layers/normalization.py]: Updated comments as below:

  # For big endian, precision difference in last decimal values getting in 
  # float32 Vs float64 data type is causing normalization_test failure. 
  # The cast to float64 will calculate mean and variance correctly while 
  # normalization of `inputs` tensor.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: tensorflow/contrib/layers/python/layers/normalization.py (hunk 1 above)
Comment (kbhute-ibm) [reply tensorflow/contrib/layers/python/layers/normalization.py]: @martinwicke Gentle reminder

**Comment by kbhute-ibm** [comment]
@martinwicke Can you please review the comments and merge the PR?

**Comment by kbhute-ibm** [comment]
Hi @martinwicke 

Please review the comments and let me know those are fine.

**Review by martinwicke — DISMISSED** [review]

**Comment by gbaned** [comment]
@kbhute-ibm Could you please address Ubuntu Sanity errors? Thanks!

**Comment by kbhute-ibm** [comment]
@gbaned Thanks for the notification. Done with fixing the Ubuntu Sanity errors.

**Comment by kbhute-ibm** [comment]
Hi @martinwicke @gbaned,

As 'contrib' folder is removed from TF 2.0 (commit: https://github.com/tensorflow/tensorflow/commit/ffc25308ce2be84240ec90502b38800bc5a4dd60) this PR has merge conflicts in master. Had few queries:
* Are we completely skipping tests present inside ./contrib from master and TF 2.0?
* Can this PR be merged for TF 1.x?

Thanks,
Koumudini

**Comment by martinwicke** [comment]
No tests for contrib will be run in TF 2.0. In fact, contrib has plainly ceased to exist outside the 1.x branches.

Can you request a cherry-pick into the 1.15 branch?

@goldiegadde FYI, this is a harmless cherrypick that we can merge into 1.15.

**Comment by kbhute-ibm** [comment]
Hi @martinwicke 
As you mentioned in the comment https://github.com/tensorflow/tensorflow/pull/31188#issuecomment-533207777, no contrib tests would run in TF 2.0. However we could still see few python tests https://github.com/tensorflow/tensorflow/tree/v2.0.0/tensorflow/contrib. The tests which were failed in earlier versions on Z do not exist in TF 2.0.

Could you please comment on this and update our knowledge?

Thanks in advance.

**Comment by martinwicke** [comment]
We haven't deleted the contrib folder on the 2.0 branch because of build
complexities. We are not running any of the tests in it though. All code
inside should be considered dead.

**Comment by kbhute-ibm** [comment]
@martinwicke Thanks for the clarification.

**Pull request closed without merging**